### PR TITLE
Issue a warning if we are falling back to the old shell environment

### LINKED ISSUE #486: Fallback to previous devshell upon failure is unclear
Since https://github.com/nix-community/nix-direnv/pull/420 (I believe), if loading the Nix devshell fails, the previous cached one is used. While I do not necessarily disagree with this behavior, I think that it should be much more clear that this is occuring, as currently one might see errors but think “well it loaded anyway, so those errors may not matter”, especially coupled with the fact that Nix *does* have non-fatal errors. Or the user might just miss those errors in a sea of other output or because they are in the flow. Either way, the shell ends up non-obviously in the “stale” state.

It may also make sense for it to be configurable whether to do this (and knowable from the outside, e.g. to condition on fallback having occurred in the `.envrc`).

### PR DESCRIPTION
**Pull request by bbenne10**
This should resolve #486.

### CODE REVIEW

File: direnvrc [hunk 1]
```diff
@@ -326,6 +326,7 @@ use_flake() {
         # which is probably more useful than nothing.
         # Fallback to use that (which means just leaving profile_rc alone!)
         _nix_direnv_warning "Evaluating current devShell failed. Falling back to previous environment!"
+        export NIX_DIRENV_DID_FALLBACK=1
```
Comment (Mic92) [inline direnvrc:329]: Who is the consumer of this environment variable?

**Comment by bbenne10** [comment]
This is expected to be exported to the calling shell for possible user consumption

-------- Original Message --------

**Comment by Mic92** [comment]
Ok. Do you have a concrete use case for it? If so it would be nice to put in the Readme.

**Comment by bbenne10** [comment]
README updated to note the environment variable. Additionally, I've changed the note about `nix-direnv-watch-file` (which is no longer accurate - oops!) and added a note about `hide_env_diff` in (what I believe to be) a more appropriate way than #485.

**Review by Mic92 — APPROVED** [review]

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**